veniq/baselines/semi/benefits.py

```diff
@@ -1,81 +1,53 @@
-from aibolit.utils.ast_builder import build_ast
-from aibolit.extract_method_baseline.extract_semantic import extract_method_statements_semantic, StatementSemantic
 
 
 def _check_is_common(
-    dict_file: Dict[int, List[str]],
-    statement_1: int,
-    statement_2: int
-) -> bool:
+        dict_file: Dict[int, List[str]],
+        statement_1: int,
+        statement_2: int
+    ) -> bool:
     '''
     This function is aimed to check whether 2 statements have
     common semantics or not.
     '''
-    joined_names: Counter = Counter(dict_file[statement_1] + dict_file[statement_2])
+
+    joined_names: Counter = Counter(defaultdict(lambda: [], dict_file)[statement_1] +
+                                    defaultdict(lambda: [], dict_file)[statement_2])
     duplicates = {element: count for element, count in joined_names.items() if count > 1}.keys()
     return len(list(duplicates)) >= 1
 
 
-def _reprocess_dict(method_semantic: Dict[ASTNode, StatementSemantic]) -> Dict[int, List[str]]:
-    reprocessed_dict = dict()
-    for statement in method_semantic.keys():
-        new_values = []
-        new_values += list(method_semantic[statement].used_variables)
-        new_values += list(method_semantic[statement].used_objects)
-        new_values += list(method_semantic[statement].used_methods)
-        reprocessed_dict[statement.line] = new_values
-    return reprocessed_dict
-
-
-def _get_dict(filepath: str) -> Dict[int, List[str]]:
-    ast = AST.build_from_javalang(build_ast(filepath))
-    classes_declarations = (
-        node for node in ast.get_root().types
-        if node.node_type == ASTNodeType.CLASS_DECLARATION
-    )
-
-    methods_declarations = (
-        method_declaration for class_declaration in classes_declarations
-        for method_declaration in class_declaration.methods
-    )
-
-    methods_ast_and_class_name = (
-        (ast.get_subtree(method_declaration), method_declaration.parent.name)
-        for method_declaration in methods_declarations
-    )
-
-    for method_ast, _ in methods_ast_and_class_name:
-        # method_name = method_ast.get_root().name
-        original_method_semantic = extract_method_statements_semantic(method_ast)
-    reprocessed_dict = _reprocess_dict(original_method_semantic)
-    return reprocessed_dict
-
-
-def _LCOM2(file_dict: Dict, range_stats=[], mode='original') -> int:
+def _LCOM2(line_to_semantic_dict: Dict,
+           range_statements: Tuple[int, int] = None,
+           mode='original') -> int:
     '''
     LCOM_2 = P - Q;
     P is the number of pairs of statements
     that do not share variables and Q is the number
-    of pairs of lines that share variables
+    of pairs of lines that share variables.
+
+    line_to_semantic_dict is a map from line number
+    to variables that are referenced there.
+    range_statements is the range of lines for which LCOM2
+    is calculated.
     '''
     P = 0
     Q = 0
     list_statements = []
 
     if mode == 'after_ref':
```

Comment:
what is `after_ref`? Consider creating ENUM structure

Reply:
it's @Vitaly-Protasov 's code
it's an indicator meaning that we consider the source code except for the range selected for extraction

We can refactor it next week